### PR TITLE
Bug 1122139 - Split common tasks to separate queues

### PR DESCRIPTION
**Pull request by maurodoglio**

### CODE REVIEW

File: treeherder/settings/base.py [hunk 1]
```diff
@@ -246,22 +264,32 @@
...
+        'options': {
+            'queue': 'process_objects'
+        }
     },
     'cycle-data-every-day': {
         'task': 'cycle-data',
         'schedule': timedelta(days=1),
-        'relative': True
+        'relative': True,
+        'queue': 'calculate_eta'
```
Comment (edmorley) [inline treeherder/settings/base.py]: This isn't inside options: {}  (unlike the others) - should it be?

Comment (maurodoglio) [inline treeherder/settings/base.py]: great catch :smile:

File: treeherder/settings/base.py [hunk 2]
```diff
@@ -246,22 +260,34 @@
...
+            'queue': 'process_objects'
+        }
     },
     'cycle-data-every-day': {
         'task': 'cycle-data',
         'schedule': timedelta(days=1),
-        'relative': True
+        'relative': True,
+        'options': {
+            'queue': 'calculate_eta'
```
Comment (edmorley) [inline treeherder/settings/base.py]: This should be 'cycle-data' :-)

File: treeherder/settings/base.py [hunk 3]
```diff
@@ -219,6 +225,14 @@
         },
         'relative': True
     },
+    'fetch-missing-push-logs-every-minute': {
+        'task': 'fetch-missing-push-logs',
+        'schedule': timedelta(minutes=1),
+        'options': {
+            "queue": "pushlog"
+        },
```
Comment (edmorley) [inline treeherder/settings/base.py]: Whilst you are here, could you move "options" after "relative", to be consistent with the others? (Or at least make them all the same, whatever order that is)

**Comment by edmorley** [comment]
Ah just spotted - could you fix the bug number in the commit message - should be bug 1122139. Thanks :-)

File: treeherder/settings/base.py [hunk 4]
```diff
@@ -195,7 +195,13 @@
     # queue for mirroring the sheriffing activity to tbpl
     Queue('high_priority', Exchange('default'), routing_key='high_priority'),
     Queue('pushlog', Exchange('default'), routing_key='pushlog'),
-    Queue('buildapi', Exchange('default'), routing_key='buildapi')
+    Queue('buildapi', Exchange('default'), routing_key='buildapi'),
+
+    Queue('process_objects', Exchange('default'), routing_key='process_objects'),
+    Queue('cycle_data', Exchange('default'), routing_key='cycle_data'),
+    Queue('calculate_eta', Exchange('default'), routing_key='calculate_eta'),
+    Queue('populate_performance_series', Exchange('default'), routing_key='populate_performance_series'),
```
Comment (edmorley) [inline treeherder/settings/base.py]: Sorry for not seeing this sooner, but I think we also need to add a routing_key here:
https://github.com/mozilla/treeherder-service/blob/master/treeherder/etl/perf_data_adapters.py#L254

**Comment by edmorley** [comment]
I think we also could do with another one here for missing pushlogs:
https://github.com/mozilla/treeherder-service/blob/master/treeherder/etl/common.py#L163

...since whilst we add it slightly later on:
https://github.com/mozilla/treeherder-service/blob/6bf711fb4a50e1bd88751b8a1d87b2ae6f789c16/treeherder/etl/tasks/cleanup_tasks.py#L33

...the initial task that splits the request into subtasks, is still run on default.

**Comment by maurodoglio** [comment]
@edmorley thanks for all the comments, much appreciated :smiley:

**Comment by edmorley** [comment]
This looks great now, thank you :-)

I think there's maybe some more tweaks we can do later (eg we have the very specific fetch_missing_push_logs and fetch_missing_hg_push_logs that could perhaps be combined, and have three tasks being put into high_priority that could probably be split into pulse_publish and classification_sync), but this gets everything off default, which is good win :-)

**Comment by edmorley** [comment]
Actually, we'll need changes to https://github.com/mozilla/treeherder-service/blob/master/bin/run_celery_worker_pushlog to also consume fetch_missing_push_logs and fetch_missing_hg_push_logs, since at the moment only pushlog is specified. Seeing as we need to make this change anyway - what do you think about combining fetch_missing_push_logs and fetch_missing_hg_push_logs ? I don't mind either way :-)

**Comment by maurodoglio** [comment]
PR updated, now fetch_missing_hg_push_logs fetch_missing_push_logs share the same queue

**Comment by edmorley** [comment]
Looks great now, thank you :-)

**Pull request merged**